### ISSUE TITLE
Feature request: docstring-min-length (pylint)

### ISSUE THREAD

**Issue by NoAchache** [astral-sh/ruff#5860]
Hello,

Would it be possible to include the pylint parameter: docstring-min-length?

It allows to only force writing docstrings if a function is bigger or equal to the specified min length.

**Comment by BradLucky** [comment]
I am really hoping this goes in to Ruff. Without it, things get very noisy since even this function will raise a warning:
```python
def sum(x, y):
    return x + y
```
It's completely self-explanatory.

**Comment by CharlesPerrotMinotHCHB** [comment]
It's also a way to disable the check of  docstrings all together ^^

**Comment by aryamccarthy** [comment]
Bumping to encourage decision from the maintainers.

**Comment by MichaReiser** [comment]
Could someone explain to me how the length is measured? I don't see an obvious definition for a length if it comes to functions

**Comment by VeckoTheGecko** [comment]
> Could someone explain to me how the length is measured?

number of lines in function body

E.g.
`t.py`
```
"""Hello"""

def sum1(x, y):
    return x + y

def sum2(x, y):
    out = x + y
    return out
```

```
pylint --docstring-min-length=1 t.py
************* Module t
t.py:3:0: C0116: Missing function or method docstring (missing-function-docstring)
t.py:6:0: C0116: Missing function or method docstring (missing-function-docstring)
```

```
pylint --docstring-min-length=2 t.py
************* Module t
t.py:6:0: C0116: Missing function or method docstring (missing-function-docstring)
```

**Comment by tjkuson** [comment]
> > Could someone explain to me how the length is measured?
> 
> number of lines in function body
> 
> E.g. `t.py`
> 
> ```
> """Hello"""
> 
> def sum1(x, y):
>     return x + y
> 
> def sum2(x, y):
>     out = x + y
>     return out
> ```
> 
> ```
> pylint --docstring-min-length=1 t.py
> ************* Module t
> t.py:3:0: C0116: Missing function or method docstring (missing-function-docstring)
> t.py:6:0: C0116: Missing function or method docstring (missing-function-docstring)
> ```
> 
> ```
> pylint --docstring-min-length=2 t.py
> ************* Module t
> t.py:6:0: C0116: Missing function or method docstring (missing-function-docstring)
> ```

Isn't this example indicative of an issue with this method compared to, for example, a complexity measure? Both `sum1` and `sum2` seem equivalent, so it would be unexpected if one required a docstring whereas the other did not.

**Comment by aryamccarthy** [comment]
If the goal is to provide feature parity for pylint users who want to transition while remaining compliant with, eg, Google’s standards, then the goodness of the complexity measure isn’t relevant.

**Comment by antspy** [comment]
Of course short functions are not necessarily less complex, but they tend to be. Or at least, there are a lot of trivial functions which do not warrant a docstring, and they tend to be short. If there are exceptions, the developer can always add a docstring if it feels like it's warranted. 

This would be quite a useful feature - otherwise docstrings become really tedious. Note that we do the same for comments - typically,  the following style of comments is discouraged: 

```python
# Sum x and y 
out = x + y 
```

Because it just adds visual clutter for absolutely no benefit. The same is true in functions docstrings, imo. 

Are you considering working on this feature?

**Comment by berkersal** [comment]
Bumping this issue as I just set up ruff for a small project and it is really amazing and I wanted to set it up for a bigger project but this literally holds me back because we don't want to document really small functions.
I would be down to try implementing it with some guidance if nobody wants to work on it
